core/src/main/scala/kafka/network/RequestConvertToJson.scala

```diff
@@ -60,7 +61,16 @@ object RequestConvertToJson {
       case req: ElectLeadersRequest => ElectLeadersRequestDataJsonConverter.write(req.data, request.version)
       case req: EndTxnRequest => EndTxnRequestDataJsonConverter.write(req.data, request.version)
       case req: EndQuorumEpochRequest => EndQuorumEpochRequestDataJsonConverter.write(req.data, request.version)
-      case req: EnvelopeRequest => EnvelopeRequestDataJsonConverter.write(req.data, request.version)
+      case req: EnvelopeRequest => {
+        val envelopeRequestData = req.data()
+        val envelopeData = envelopeRequestData.requestData().duplicate()
+        val envelopeHeader = EnvelopeUtils.parseForwardedRequestHeader(envelopeData)
+        val requestJson = RequestConvertToJson.request(AbstractRequest.parseRequest(envelopeHeader.apiKey(), envelopeHeader.apiVersion(), envelopeData).request)
```

Comment:
Hey @mumrah, does this parse the enveloped request twice? If so, is it possible to implement this feature without parsing the enveloped request twice?